tests/Mailer/MailerTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\NewsBundle\Tests\Mailer;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Sonata\NewsBundle\Mailer\Mailer;
+use Sonata\NewsBundle\Model\BlogInterface;
+use Sonata\NewsBundle\Model\CommentInterface;
+use Sonata\NewsBundle\Model\PostInterface;
+use Sonata\NewsBundle\Util\HashGeneratorInterface;
+use Symfony\Component\Mailer\MailerInterface as SymfonyMailerInterface;
+use Symfony\Component\Mime\Email;
+use Symfony\Component\Routing\RouterInterface;
+use Symfony\Component\Templating\EngineInterface;
+
+class MailerTest extends TestCase
+{
+    /**
+     * @var RouterInterface|MockObject
+     */
+    private $router;
+
+    /**
+     * @var BlogInterface|MockObject
+     */
+    private $blog;
+
+    /**
+     * @var HashGeneratorInterface|MockObject
+     */
+    private $generator;
+
+    /**
+     * @var EngineInterface|MockObject
+     */
+    private $templating;
+
+    /**
+     * @var Mailer
+     */
+    private $mailer;
+
+    /**
+     * @var array
+     */
+    private $emails;
+
+    protected function setUp(): void
+    {
+        $this->mailer = $this->createMock(SymfonyMailerInterface::class);
```

Comment:
The docblock at `MailerTest::$mailer` is only allowing instances of  `Sonata\NewsBundle\Mailer\Mailer`.

Reply:
Fixed